### PR TITLE
Update docs and tox

### LINKED ISSUE #683: Fix references to version 3.0 in docstring of SQLAlchemy
This in in the 2.3 maintenance branch, so I assume these changes were released earlier than 3.0:

```
    .. versionadded:: 3.0
       The `query_class` parameter was added, to allow customisation
       of the query class, in place of the default of :class:`BaseQuery`.

       The `model_class` parameter was added, which allows a custom model
       class to be used in place of :class:`Model`.

    .. versionchanged:: 3.0
       Utilise the same query class across `session`, `Model.query` and `Query`.
```

### LINKED ISSUE #686: Tox docs build failing
At least on the 2.x branch, see #684:

> docs_html: /home/travis/build/pallets/flask-sqlalchemy/.tox/docs_html/lib/python3.6/site-packages/flask_sqlalchemy/init.py:docstring of flask_sqlalchemy.SignallingSession.get_bind:63:Unknown
interpreted text role "paramref".

### PR DESCRIPTION
**Pull request by rsyring**
- refs #685:
  - dropping support in CI for older Python versions
  - adding CI support for Python3.7
  - refactoring tox/CI as needed
- fixes #686 - tox docs build failing
- fixes #683 - fix references to version 3.0 in SQLAlchemy docstring
- improves readme and documentation including better checks in CI
- Allow to pass args to pytest through tox - replaces #696

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
+Flask-SQLAlchemy
```
Comment (davidism) [inline README.rst]: Use the same style readme as Flask please. I'm trying to be consistent across Pallets projects.

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+################
+
+.. image:: https://travis-ci.org/pallets/flask-sqlalchemy.svg?branch=2.x-maintenance
```
Comment (davidism) [inline README.rst]: I prefer not to have badges in the Pallets readmes. I find them more distracting than helpful, and the relevant links are at the bottom already.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,105 @@
+Flask-SQLAlchemy
+################
```
Comment (davidism) [inline README.rst]: Pallets style is `==` for first level header, `-` for second.

**Comment by davidism** [comment]
You can merge this, especially just to get the tests passing, but I want to go through and do a pass to get everything consistent with the other Pallets projects before release.

**Comment by rsyring** [comment]
I'll go ahead and update the readme styles to match Flask and then merge.  Are the styles documented anywhere or is it more of a "just match the other projects" type of thing?

File: README.rst (hunk 2 above)
Comment (rsyring) [reply README.rst]: I actually find them really helpful, but I'm happy to defer here.  :)  I'll remove them.

**Comment by davidism** [comment]
Please remember to squash intermediate commits before merging.

**Pull request merged**